### PR TITLE
Add type to style and overlayStyle

### PR DESCRIPTION
**Pull request by maryjenel**
# Overview
Received this error in `node_modules` 

```Error ┈┈┈┈┈ node_modules/react-native-share/index.js:73:52

Cannot create View element because inexact object type [1]
is incompatible with exact $ReadOnly [2] in array element
of property style.

     node_modules/react-native-share/index.js
 [1]  45│   overlayStyle?: {},
        :
      70│     const {style = {}, overlayStyle = {}, ...props} = this.props;
      71│     return (
      72│       <Overlay visible={this.props.visible} {...props}>
      73│         <View style={[styles.actionSheetContainer, overlayStyle]}>
      74│           <TouchableOpacity
      75│             style={shareSheetStyle}
      76│             onPress={this.props.onCancel}
      77│           />
      78│           <Sheet visible={this.props.visible}>
      79│             <View style={[styles.buttonContainer, style]}>
      80│               {this.props.children}
      81│             </View>
      82│           </Sheet>
      83│         </View>
      84│       </Overlay>
      85│     );
      86│   }

     node_modules/react-native/Libraries/StyleSheet/StyleSheetTypes.js
 [2] 647│   $ReadOnly<$Shape<____ViewStyle_Internal>>,


Error ┈┈┈┈┈ node_modules/react-native-share/index.js:79:51

Cannot create View element because inexact object type [1]
is incompatible with exact $ReadOnly [2] in array element
of property style.

     node_modules/react-native-share/index.js
 [1]  44│   style?: {},
        :
      76│             onPress={this.props.onCancel}
      77│           />
      78│           <Sheet visible={this.props.visible}>
      79│             <View style={[styles.buttonContainer, style]}>
      80│               {this.props.children}
      81│             </View>
      82│           </Sheet>
      83│         </View>
      84│       </Overlay>

     node_modules/react-native/Libraries/StyleSheet/StyleSheetTypes.js
 [2] 647│   $ReadOnly<$Shape<____ViewStyle_Internal>>,```
# Test Plan
Implement flow in project
run `yarn flow`
observe.

### CODE REVIEW

**Comment by MateusAndrade** [comment]
LGTM! You tested it on a project using rn-share @maryjenel? 

Btw, thanks for your help! 😄

File: index.js [hunk 1]
```diff
@@ -41,8 +42,8 @@ type Props = {
   visible: boolean,
   onCancel: () => void,
   children: React.Node,
-  style?: {},
-  overlayStyle?: {},
+  style?: $ReadOnlyArray<ViewStyleProp>,
```
Comment (charpeni) [inline index.js]:
```suggestion
  style?: ViewStyleProp,
```

File: index.js [hunk 2]
```diff
@@ -41,8 +42,8 @@ type Props = {
...
+  overlayStyle?: $ReadOnlyArray<ViewStyleProp>,
```
Comment (charpeni) [inline index.js]:
```suggestion
  overlayStyle?: ViewStyleProp,
```

**Comment by charpeni** [comment]
> CircleCI (and our commit hook I think?) run a flow analysis so it passed that, and seems okay to me.

Good catch! #539 aims to fix this.

> would say this will break existing installed users, or is this type backwards compatible?

Worst case scenario is that users will now have a legit flow error saying that their style is not compatible with `ViewStyleProp`.

> You tested it on a project using rn-share @maryjenel?

I'll answer for her as a colleague 👋, saying that we tested it into our application. After upgrading to RN 0.59, we started having those flow errors, and we solved it like [the suggested changes](https://github.com/react-native-community/react-native-share/pull/538#pullrequestreview-257760677).

Those changes make sense based on RN's flow types [here](https://github.com/facebook/react-native/blob/4a5a90926447611b6ff631820b21ed07be3423ed/Libraries/StyleSheet/StyleSheet.js#L30-L38), instead of relying on a plain object.

**Comment by mikehardy** [comment]
@MateusAndrade @jgcmarins - I’m going to be traveling in remote locations the next few weeks, so I won’t be able to review more, sorry 🙏 , but I’ll be back at it in August

**Comment by MateusAndrade** [comment]
> @MateusAndrade @jgcmarins - I’m going to be traveling in remote locations the next few weeks, so I won’t be able to review more, sorry  , but I’ll be back at it in August

Have a Nice trip. :smile:  You are going to visita São Paulo? Hahahaha

Btw, im finishing some things on my daily job this week. After that, im going to focus on the os projects. :D

**Comment by mikehardy** [comment]
Hopefully Brazil soon, mostly Bolivia this time, para hacer turismo. Brazil dentro de poco para negocio :-). First I guess I should learn Portuguese instead of just Spanish though :laughing:

**Pull request merged**